drivers/overlay/ov_network.go

```diff
@@ -109,6 +109,11 @@ func (d *driver) DeleteNetwork(nid string) error {
 		return fmt.Errorf("could not find network with id %s", nid)
 	}
 
+	// Make sure driver resources are initialized before proceeding
+	if err := d.configure(); err != nil {
```

Comment:
You should do this before trying to get `d.network` because it may depend on the configuration to be completed.